hack/start-dev-env-dynamicfile.sh

```diff
@@ -56,6 +56,7 @@ write_kind_config
 create_kind_cluster
 certificate_authority_data="$(extract_certificate_authority_data)"
 write_kubectl_kubeconfig
+echo "start end to end testing for mountfile mode"
```

Comment:
My only nit is this is called "start dev env" but its used to run e2e tests.  Do you use this for both?  I'd like to separate them and have e2e tests run from the e2e script and dev envs set up from the dev script, and just have options to choose between dynamic file or other backends.  We can do that in a future improvement if you'd like.

Reply:
thanks @nckturner! this is a good point. I will create a separate PR after this cherry-pick to move the e2e out.